src/Subscriber/LoggerSubscriber.php

```diff
@@ -0,0 +1,137 @@
+ *
+ * If only includedEntities is empty,
+ * everything will be logged unless the processed entity is
+ * an instanceof OR is_subclass_of at least one element of the exclude list.
+ *
+ * If only excludeEntities is empty,
+ * only the entities instanceof OR is_subclass_of at least one element of the include list will be logged.
+ *
+ * If both lists are not empty,
+ * the entity has to be an instanceof OR is_subclass_of at least one element of the include list
+ * AND NOT an instanceof or is_subclass_of at least one element of the exclude list.
+ */
+class LoggerSubscriber implements EventSubscriber
+{
+    private LogSerializer $formatter;
+    private LogFactoryInterface $factory;
+    private array $includedEntities;
+    private array $excludedEntities;
+
+    public function __construct(
+        LogSerializer $formatter,
+        LogFactoryInterface $factory,
+        array $includedEntities,
+        array $excludedEntities
+    ) {
+        $this->formatter = $formatter;
+        $this->factory = $factory;
+        $this->includedEntities = $includedEntities;
+        $this->excludedEntities = $excludedEntities;
+    }
+
+    public function getSubscribedEvents()
+    {
+        return [Events::onFlush];
+    }
+
+    public function onFlush(OnFlushEventArgs $eventArgs): void
+    {
+        $entityManager = $eventArgs->getEntityManager();
+        $unitOfWork = $entityManager->getUnitOfWork();
+
+        $logs = [];
+
+        //Creation
+        foreach ($unitOfWork->getScheduledEntityInsertions() as $entity) {
+            if ($this->isLoggeable($entity)) {
+                $logs[] = $this->factory->createLogFromEntity($entity, 'action.create');
+            }
+        }
+
+        //Update
+        foreach ($unitOfWork->getScheduledEntityUpdates() as $entity) {
+            if ($this->isLoggeable($entity)) {
+                $logs = array_merge($logs, $this->getLogsForEntityFields($entity, $entityManager));
+            }
+        }
+
+        //Delete
+        foreach ($unitOfWork->getScheduledEntityDeletions() as $entity) {
+            if ($this->isLoggeable($entity)) {
+                $logs[] = $this->factory->createLogFromEntity($entity, 'action.delete');
+            }
+        }
+
+        $cmf = $entityManager->getMetadataFactory();
+
+        foreach ($logs as $log) {
+            $entityManager->persist($log);
+            $unitOfWork->computeChangeSet($cmf->getMetadataFor(get_class($log)), $log);
+        }
+    }
+
+    private function isLoggeable($entity): bool
+    {
+        return (empty($this->includedEntities) || $this->isSubClassFromList($entity, $this->includedEntities))
+            && (empty($this->excludedEntities) || !$this->isSubClassFromList($entity, $this->excludedEntities));
+    }
+
+    private function isSubClassFromList($entity, array $classes): bool
+    {
+        foreach ($classes as $class) {
+            if ($entity instanceof $class || is_subclass_of($entity, $class)) {
```

Comment:
On pourrait pas simplifier avec is_a ?
https://www.php.net/manual/fr/function.is-a.php